parsely/src/test/java/com/parsely/parselyandroid/HeartbeatIntervalCalculatorTest.kt

```diff
@@ -0,0 +1,82 @@
+package com.parsely.parselyandroid
+
+import com.parsely.parselyandroid.HeartbeatIntervalCalculator.Companion.MAX_TIME_BETWEEN_HEARTBEATS
+import java.util.Calendar
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+
+internal class HeartbeatIntervalCalculatorTest {
+
+    private lateinit var sut: HeartbeatIntervalCalculator
+    private val fakeClock = FakeClock()
+
+    @Before
+    fun setUp() {
+        sut = HeartbeatIntervalCalculator(fakeClock)
+    }
+
+    @Test
+    fun `given the same time of start and current time, when calculating interval, return offset times backoff proportion`() {
+        // given
+        fakeClock.fakeNow = Duration.ZERO
+        val startTime = Calendar.getInstance().apply {
+            timeInMillis = 0
+        }
+
+        // when
+        val result = sut.calculate(startTime)
+
+        // then
+        // ((currentTime + offset) * backoff) and then in milliseconds
+        // (0 + 35) * 0.3 * 1000 = 10500
+        assertThat(result).isEqualTo(10500)
+    }
+
+    @Test
+    fun `given a time that will cause the interval to surpass the MAX_TIME_BETWEEN_HEARTBEATS, when calculating interval, then return the MAX_TIME_BETWEEN_HEARTBEATS`() {
+        // given
+        // "excessiveTime" is a calculated point in time where the resulting interval would
+        // surpass MAX_TIME_BETWEEN_HEARTBEATS
+        // (currentTime + offset) * backoff = max
+        // currentTime = (max / backoff) - offset, so
+        // (15 minutes / 0.3) - 35 seconds = 2965 seconds. Add 1 second to be over the limit
```

Comment:
Praise (❤️): Thanks for adding these comments and making it clear how these numbers are calculated. 🙇